include/EntryPointConfirmOptInHandler.php

```diff
@@ -161,11 +162,49 @@ private function methodConfirmOptInUser($request)
         if ($this->emailAddress) {
             $this->emailAddress->confirmOptIn();
             $this->emailAddress->save();
-        }
 
+            $people = $this->getIDs($this->emailAddress->email_address, 'Contacts');
+            if ($people) {
+                $this->setLawfulBasisForEachPerson($people, 'Contacts');
+            }
+            $people = $this->getIDs($this->emailAddress->email_address, 'Leads');
+            if ($people) {
+                $this->setLawfulBasisForEachPerson($people, 'Leads');
+            }
+
+            $people = $this->getIDs($this->emailAddress->email_address, 'Prospects');
+            if ($people) {
+                $this->setLawfulBasisForEachPerson($people,  'Prospects');
+            }
+        }
         $template = new Sugar_Smarty();
         $template->assign('FOCUS', $this->emailAddress);
 
         return $template->fetch('include/EntryPointConfirmOptIn.tpl');
     }
+
+    /**
+     * @param String $email
+     * @param String $module
+     *
+     * @return array|bool
+     */
+    private function getIDs($email, $module) {
+        $people = $this->emailAddress->getRelatedId($email, $module);
+        return $people;
+    }
+
+    /**
+     * @param array $people
+     */
+    private function setLawfulBasisForEachPerson(array $people, $module) {
+        /** @var Person $person */
+        foreach ($people as $person) {
+            $bean = BeanFactory::getBean($module, $person);
+            if($bean) {
+                $bean->setLawfulBasis('consent', 'email');
+                $bean->save();
```

Comment:
Please handle if $bean-save() failing. for e.g an Error log or user message. btw. the setLawFulBasis() method also save the bean at the end?

Reply:
Agreed - that is a double saving...